api/src/services/bcgw-layer-service.ts

```diff
@@ -0,0 +1,460 @@
+   * without running into any inconsistent layer issues (ex: NRM Region 1 should map to ENV Region 2, but due to
+   * the polygons in the layers overlapping, NRM Region 1 will return matches for ENV Region 2 AND ENV Region 3, when
+   * it shouldn't). As a result we cannot reply fully on the layer query results alone.
+   *
+   * @param {RegionDetails[]} regionDetails
+   * @return {*}
+   * @memberof BcgwLayerService
+   */
+  getMappedRegionDetails(regionDetails: RegionDetails[]) {
+    if (!regionDetails?.length) {
+      // feature is missing any identifying attributes
+      return [];
+    }
+
+    const response = [...regionDetails];
+
+    for (const regionDetail of regionDetails) {
+      // Check if the regionDetail item came from the ENV layer
+      if (regionDetail.sourceLayer === BcgwEnvRegionsLayer) {
+        // Check if the region is a known ENV region that maps to a known NRM region
+        const matchingNrmRegion = envToNrmRegionsMapping[regionDetail.regionName];
+        if (matchingNrmRegion) {
+          // Add NRM region to response
+          response.push({ regionName: matchingNrmRegion, sourceLayer: BcgwNrmRegionsLayer });
+        }
+      }
+
+      // Check if the regionDetail item came from the NRM layer
+      if (regionDetail.sourceLayer === BcgwNrmRegionsLayer) {
+        // Check if the region is a known NRM region that maps to one or more known ENV regions
+        Object.entries(envToNrmRegionsMapping).forEach(([envRegion, nrmRegion]) => {
+          if (regionDetail.regionName === nrmRegion) {
+            // Return matching ENV region
+            response.push({ regionName: envRegion, sourceLayer: BcgwEnvRegionsLayer });
+          }
+        });
+      }
+    }
+
+    return response;
+  }
+
+  /**
+   * For a given GeoJSON Feature, fetch all region details from all supported BCGW layers.
+   *
+   * @param {Feature} feature
+   * @param {IDBConnection} connection
+   * @return {*}
+   * @memberof BcgwLayerService
+   */
+  async getRegionsForFeature(feature: Feature, connection: IDBConnection) {
+    // Array of all matching region details for the feature
+    let response: RegionDetails[] = [];
+
+    const postgisService = new PostgisService(connection);
+    // Convert the feature geometry to WKT format
+    const result = await postgisService.getGeoJsonGeometryAsWkt(feature.geometry, Srid3005);
+    const geometryWKTString = result.geometry;
+
+    // Attempt to detect if the feature is a known BCGW feature
+    const regionDetails = this.findRegionDetails(feature);
+
+    if (!regionDetails) {
+      // Feature is not a known BCGW feature
+      // Fetch region details for the feature from ALL available layers
+      response = response.concat(
+        await this.getAllRegionDetailsForWktString(geometryWKTString, [
+          BcgwEnvRegionsLayer,
+          BcgwNrmRegionsLayer
+          // BcgwParksAndEcoreservesLayer,
```

Comment:
Any reason these commented lines should stay?

Reply:
Yeah, its because I was bouncing between using those layers or not. They was support for them in the frontend, but evidently it was disable or something. I moved that all to the API. But when the matching parks are returned, you get a massive list (like 40 items) if the boundary is an NRM region. So it seemed like that wasn't going to be helpful, even though it technically works, so I just commented them out. But, could remove them fully probably.

Reply:
Removed.